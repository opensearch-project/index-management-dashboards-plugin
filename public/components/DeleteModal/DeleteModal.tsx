/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from "react";
import {
  EuiSmallButton,
  EuiSmallButtonEmpty,
  EuiCompressedFieldText,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiSpacer,
  EuiText,
} from "@elastic/eui";

interface DeleteModalProps {
  selectedItems: string[];
  visible: boolean;
  onClose: () => void;
  onConfirm: () => void;
  title: string;
  tips: string;
}

export default function DeleteTemplateModal(props: DeleteModalProps) {
  const [value, setValue] = useState("");
  const { onClose, visible, selectedItems, onConfirm } = props;
  useEffect(() => {
    if (visible) {
      setValue("");
    }
  }, [visible]);

  if (!visible) {
    return null;
  }

  return (
    <EuiModal onClose={onClose}>
      <EuiModalHeader>
        <EuiModalHeaderTitle>
          <EuiText size="s">
            <h2>{props.title}</h2>
          </EuiText>
        </EuiModalHeaderTitle>
      </EuiModalHeader>

      <EuiModalBody>
        <div style={{ lineHeight: 1.5 }}>
          <EuiText size="s">
            <p>{props.tips}</p>
          </EuiText>
          <ul style={{ listStyleType: "disc", listStylePosition: "inside" }}>
            {selectedItems.map((item) => (
              <EuiText size="s">
                <li key={item}>{item}</li>
              </EuiText>
            ))}
          </ul>
          <EuiSpacer />
          <EuiText color="subdued" size="s">
            <p>
              To confirm your action, type <b style={{ color: "#000" }}>delete</b>.
            </p>
          </EuiText>
          <EuiCompressedFieldText
            data-test-subj="deleteInput"
            placeholder="delete"
            fullWidth
            value={value}
            onChange={(e) => setValue(e.target.value)}
          />
        </div>
      </EuiModalBody>

      <EuiModalFooter>
        <EuiSmallButtonEmpty data-test-subj="deletaCancelButton" onClick={onClose}>
          Cancel
        </EuiSmallButtonEmpty>
        <EuiSmallButton data-test-subj="deleteConfirmButton" onClick={onConfirm} fill color="danger" disabled={value !== "delete"}>
          Delete
        </EuiSmallButton>
      </EuiModalFooter>
    </EuiModal>
  );
}
