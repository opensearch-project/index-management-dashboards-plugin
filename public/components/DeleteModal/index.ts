/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import DeleteTemplateModal from "./DeleteModal";

export default DeleteTemplateModal;
