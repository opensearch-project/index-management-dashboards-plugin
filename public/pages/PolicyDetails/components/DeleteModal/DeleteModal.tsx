/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React, { ChangeEvent, Component, Fragment } from "react";
import { EuiConfirmModal, EuiForm, EuiFormRow, EuiFieldText, EuiOverlayMask, EuiSpacer } from "@elastic/eui";

interface DeleteModalProps {
 policyId: string;
 closeDeleteModal: (event?: any) => void;
 onClickDelete: (event?: any) => void;
}

interface DeleteModalState {
 confirmDeleteText: string;
}

export default class DeleteModal extends Component<DeleteModalProps, DeleteModalState> {
 state = { confirmDeleteText: "" };

 onChange = (e: ChangeEvent<HTMLInputElement>): void => {
   this.setState({ confirmDeleteText: e.target.value });
 };

 render() {
   const { policyId, closeDeleteModal, onClickDelete } = this.props;
   const { confirmDeleteText } = this.state;

   return (
     <EuiOverlayMask>
       <EuiConfirmModal
         title="Delete policy"
         onCancel={closeDeleteModal}
         onConfirm={onClickDelete}
         cancelButtonText="Cancel"
         confirmButtonText="Delete policy"
         buttonColor="danger"
         defaultFocusedButton="confirm"
         confirmButtonDisabled={confirmDeleteText != "delete"}
       >
         <EuiForm>
           <Fragment>
             Delete "<strong>{policyId}</strong>" permanently? Indices will no
             longer be managed using this policy
           </Fragment>
           <EuiSpacer size="s" />
           <EuiFormRow helpText="To confirm deletion, type delete">
             <EuiFieldText value={confirmDeleteText} placeholder="delete" onChange={this.onChange} data-test-subj="deleteTextField" />
           </EuiFormRow>
         </EuiForm>
       </EuiConfirmModal>
     </EuiOverlayMask>
   );
 }
}
