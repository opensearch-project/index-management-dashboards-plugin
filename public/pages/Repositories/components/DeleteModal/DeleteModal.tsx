/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { ChangeEvent, Component } from "react";
import { EuiConfirmModal, EuiCompressedFieldText, EuiForm, EuiCompressedFormRow, EuiOverlayMask, EuiSpacer, EuiText } from "@elastic/eui";

interface DeleteModalProps {
  closeDeleteModal: (event?: any) => void;
  onClickDelete: (event?: any) => void;
  type: string;
  ids: string;
  addtionalWarning?: string;
  confirmation?: boolean;
}

interface DeleteModalState {
  confirmDeleteText: string;
}

export default class DeleteModal extends Component<DeleteModalProps, DeleteModalState> {
  constructor(props: DeleteModalProps) {
    super(props);

    let confirmDeleteText = "delete";
    if (props.confirmation) confirmDeleteText = "";
    this.state = {
      confirmDeleteText,
    };
  }

  onChange = (e: ChangeEvent<HTMLInputElement>): void => {
    this.setState({ confirmDeleteText: e.target.value });
  };

  render() {
    const { type, ids, closeDeleteModal, onClickDelete, addtionalWarning, confirmation } = this.props;
    const { confirmDeleteText } = this.state;

    return (
      <EuiOverlayMask>
        <EuiConfirmModal
          title={
            <EuiText size="s">
              <h2>Delete {type}</h2>
            </EuiText>
          }
          onCancel={closeDeleteModal}
          onConfirm={() => {
            onClickDelete();
            closeDeleteModal();
          }}
          cancelButtonText="Cancel"
          confirmButtonText={`Delete ${type}`}
          buttonColor="danger"
          defaultFocusedButton="confirm"
          confirmButtonDisabled={confirmDeleteText != "delete"}
        >
          <EuiForm>
            <p>
              Delete "<strong>{ids}</strong>" permanently? {addtionalWarning}
            </p>
            <EuiSpacer size="s" />
            {!!confirmation && (
              <EuiCompressedFormRow helpText={`To confirm deletion, type "delete".`}>
                <EuiCompressedFieldText
                  value={confirmDeleteText}
                  placeholder="delete"
                  onChange={this.onChange}
                  data-test-subj="deleteTextField"
                />
              </EuiCompressedFormRow>
            )}
          </EuiForm>
        </EuiConfirmModal>
      </EuiOverlayMask>
    );
  }
}
