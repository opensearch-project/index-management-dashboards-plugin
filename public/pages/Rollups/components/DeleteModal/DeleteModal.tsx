/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { ChangeEvent, Component, Fragment } from "react";
import { EuiConfirmModal, EuiForm, EuiCompressedFormRow, EuiCompressedFieldText, EuiOverlayMask, EuiSpacer, EuiText } from "@elastic/eui";

interface DeleteModalProps {
  rollupId: string;
  closeDeleteModal: (event?: any) => void;
  onClickDelete: (event?: any) => void;
}

interface DeleteModalState {
  confirmDeleteText: string;
}

export default class DeleteModal extends Component<DeleteModalProps, DeleteModalState> {
  state = { confirmDeleteText: "" };

  onChange = (e: ChangeEvent<HTMLInputElement>): void => {
    this.setState({ confirmDeleteText: e.target.value });
  };

  render() {
    const { rollupId, closeDeleteModal, onClickDelete } = this.props;
    const { confirmDeleteText } = this.state;

    return (
      <EuiOverlayMask>
        <EuiConfirmModal
          title={
            <EuiText size="s">
              <h2>Delete job(s)</h2>
            </EuiText>
          }
          onCancel={closeDeleteModal}
          onConfirm={onClickDelete}
          cancelButtonText="Cancel"
          confirmButtonText="Delete"
          buttonColor="danger"
          defaultFocusedButton="confirm"
          confirmButtonDisabled={confirmDeleteText != "delete"}
        >
          <EuiForm>
            <Fragment>
              <EuiText size="s">
                By deleting "<strong>{rollupId}</strong>", all future scheduled rollup execution will be canceled. However, your target
                index will remain as it is.
              </EuiText>
            </Fragment>
            <EuiSpacer size="s" />
            <EuiCompressedFormRow helpText="To confirm deletion, enter delete in the text field">
              <EuiCompressedFieldText
                value={confirmDeleteText}
                placeholder="delete"
                onChange={this.onChange}
                data-test-subj="deleteTextField"
              />
            </EuiCompressedFormRow>
          </EuiForm>
        </EuiConfirmModal>
      </EuiOverlayMask>
    );
  }
}
