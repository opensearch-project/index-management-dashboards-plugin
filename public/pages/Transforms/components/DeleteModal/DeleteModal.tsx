/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { ChangeEvent, Component, Fragment } from "react";
import { EuiConfirmModal, EuiForm, EuiCompressedFormRow, EuiCompressedFieldText, EuiOverlayMask, EuiSpacer, EuiText } from "@elastic/eui";

// TODO: Merge with Rollup to create generic component
interface DeleteModalProps {
  item: string;
  closeDeleteModal: (event?: any) => void;
  onClickDelete: (event?: any) => void;
}

interface DeleteModalState {
  confirmDeleteText: string;
}

export default class DeleteModal extends Component<DeleteModalProps, DeleteModalState> {
  state = { confirmDeleteText: "" };

  onChange = (e: ChangeEvent<HTMLInputElement>): void => {
    this.setState({ confirmDeleteText: e.target.value });
  };

  render() {
    const { item, closeDeleteModal, onClickDelete } = this.props;
    const { confirmDeleteText } = this.state;

    return (
      <EuiOverlayMask>
        <EuiConfirmModal
          title={
            <EuiText size="s">
              <h2>Delete job(s)</h2>
            </EuiText>
          }
          onCancel={closeDeleteModal}
          onConfirm={onClickDelete}
          cancelButtonText="Cancel"
          confirmButtonText="Delete"
          buttonColor="danger"
          defaultFocusedButton="confirm"
          confirmButtonDisabled={confirmDeleteText != "delete"}
        >
          <EuiForm>
            <Fragment>
              <EuiText size="s">
                <p>
                  By deleting <strong>{item}</strong>, all future scheduled executions will be canceled. However, your target index and data
                  in it will remain intact.
                </p>
              </EuiText>
            </Fragment>
            <EuiSpacer size="s" />
            <EuiCompressedFormRow helpText="To confirm deletion, enter delete in the text field">
              <EuiCompressedFieldText
                value={confirmDeleteText}
                placeholder="delete"
                onChange={this.onChange}
                data-test-subj="deleteTextField"
              />
            </EuiCompressedFormRow>
          </EuiForm>
        </EuiConfirmModal>
      </EuiOverlayMask>
    );
  }
}
