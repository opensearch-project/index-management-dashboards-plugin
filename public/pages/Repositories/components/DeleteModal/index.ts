/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import DeleteModal from "./DeleteModal";
export default DeleteModal;
