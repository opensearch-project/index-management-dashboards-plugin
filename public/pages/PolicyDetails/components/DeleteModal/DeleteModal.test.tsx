/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { screen, render } from "@testing-library/react";
import DeleteModal from "./DeleteModal";
import { fireEvent } from "@testing-library/dom";
import userEvent from "@testing-library/user-event/dist";

describe("<DeleteModal /> spec", () => {
  it("renders the component", () => {
    const { baseElement } = render(
      <DeleteModal
        policyId="some_id"
        closeDeleteModal={() => {}}
        onClickDelete={() => {}}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("calls closeDeleteModal when cancel button is clicked", () => {
    const closeDeleteModal = jest.fn();
    const { getByTestId } = render(
      <DeleteModal
        policyId="some_id"
        closeDeleteModal={closeDeleteModal}
        onClickDelete={() => {}}
      />
    );

    userEvent.click(getByTestId("confirmModalCancelButton"));
    expect(closeDeleteModal).toHaveBeenCalled();
  });

  it("calls onClickDelete when delete button is clicked", () => {
    const onClickDelete = jest.fn();
    const { getByTestId } = render(
      <DeleteModal
        policyId="some_id"
        closeDeleteModal={() => {}}
        onClickDelete={onClickDelete}
      />
    );

    fireEvent.focus(getByTestId("deleteTextField"));
    userEvent.type(getByTestId("deleteTextField"), `delete`);
    userEvent.click(getByTestId("confirmModalConfirmButton"));

    expect(onClickDelete).toHaveBeenCalled();
  });
});
