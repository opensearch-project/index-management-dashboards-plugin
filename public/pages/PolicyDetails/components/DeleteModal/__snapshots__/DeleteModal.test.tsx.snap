// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteModal /> spec renders the component 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default euiModal--confirmation"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
              data-test-subj="confirmModalTitleText"
            >
              Delete policy
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                class="euiText euiText--medium"
                data-test-subj="confirmModalBodyText"
              >
                <div
                  class="euiForm"
                >
                  Delete "
                  <strong>
                    some_id
                  </strong>
                  " permanently? Indices will no longer be managed using this policy
                  <div
                    class="euiSpacer euiSpacer--s"
                  />
                  <div
                    class="euiFormRow"
                    id="some_html_id-row"
                  >
                    <div
                      class="euiFormRow__fieldWrapper"
                    >
                      <div
                        class="euiFormControlLayout"
                      >
                        <div
                          class="euiFormControlLayout__childrenWrapper"
                        >
                          <input
                            aria-describedby="some_html_id-help"
                            class="euiFieldText"
                            data-test-subj="deleteTextField"
                            id="some_html_id"
                            placeholder="delete"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="euiFormHelpText euiFormRow__text"
                        id="some_html_id-help"
                      >
                        To confirm deletion, type delete
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              data-test-subj="confirmModalCancelButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--danger euiButton--fill euiButton-isDisabled"
              data-test-subj="confirmModalConfirmButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Delete policy
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;
